spack/packages/dla-future/package.py

```diff
@@ -84,12 +84,11 @@ def cmake_args(self):
         # CUDA/HIP
         args.append(self.define_from_variant("DLAF_WITH_CUDA", "cuda"))
         args.append(self.define_from_variant("DLAF_WITH_HIP", "rocm"))
-
-        # HIP support requires compiling with hipcc
-        if '+rocm' in self.spec:
-            args += [self.define('CMAKE_CXX_COMPILER', self.spec['hip'].hipcc)]
-            if self.spec.satisfies('^cmake@3.21.0:3.21.2'):
-                args += [self.define('__skip_rocmclang', True)]
+        if "+rocm" in spec:
+            archs = self.spec.variants["amdgpu_target"].value
+            if archs != "none":
+                arch_str = ";".join(archs)
+                args.append(self.define("CMAKE_HIP_ARCHITECTURES", arch_str))
```

Comment:
See https://cmake.org/cmake/help/latest/variable/CMAKE_HIP_ARCHITECTURES.html?highlight=cmake_hip_architectures. Basically if one doesn't do the above to set `CMAKE_HIP_ARCHITECTURES` in spack the `amdgpu_target` variant has no effect as @aurianer pointed out. The same applies for CUDA.

Generally I consider `CMAKE_HIP/CUDA_ARCHITECTURES` as required variables, and if they're not set I'd prefer to use whatever default the CMake modules decide on rather than hardcoding static defaults in DLA-Future's CMake.

Reply:
Thanks for the replies!